### PR TITLE
feat: Adding a new no-ignored-unsubscribe rule.

### PR DESCRIPTION
**Pull request by moufmouf**
Hey @ota-meshi

First of all, thanks for this package, it's really a life saver!

I would like here to propose a new rule for the eslint Svelte plugin: `no-ignored-unsubscribe`.

This rule fails if an "unsubscriber" returned by call to `subscribe()` is neither assigned to a variable or property nor passed to a function.

One should always unsubscribe from a store when it is no longer needed. Otherwise, the subscription will remain active and constitute a **memory leak**. This rule helps to find such cases by ensuring that the unsubscriber (the return value from the store's `subscribe` method) is not ignored.

I'm actually writing this rule after being bite by such an issue => https://github.com/workadventure/workadventure/pull/3527/files

About the code:

The rule is heavily inspired by a similar rule in eslint-plugin-rxjs: https://github.com/cartant/eslint-plugin-rxjs/blob/main/docs/rules/no-ignored-subscription.md

I did simplify the code a bit. As it is now, the rule will generate a few false positives (it is triggered on all method named `subscribe`, whether the underlying object is a store or not). In practice, I think this is acceptable.

There is alas no simple way to check the type of the underlying object. The RxJS rule relies on analyzing the code to try to find the type of the object. I tested a similar approach. It has a number of false negatives, and the code is really more complex as it relies on doing some type testing with typescript-eslint (as described [here](https://typescript-eslint.io/packages/rule-tester/#type-aware-testing)).

Let me know what you think!

### CODE REVIEW

File: src/rules/no-ignored-unsubscribe.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import type { TSESTree } from '@typescript-eslint/types';
+import { createRule } from "../utils";
+
+export default createRule("no-ignored-unsubscribe", {
+	meta: {
+		docs: {
+			description: "Forbids ignoring the unsubscribe method returned by the `subscribe()` os Svelte stores.",
```
Comment (ota-meshi) [inline src/rules/no-ignored-unsubscribe.ts:7]: Start with `disallow` according to the documentation. Also, there's probably a typo.

```suggestion
			description: "disallow ignoring the unsubscribe method returned by the `subscribe()` on Svelte stores",
```

File: src/rules/no-ignored-unsubscribe.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		hasSuggestions: false,
+		messages: {
+			forbidden: "Ignoring returned value of the subscribe method is forbidden.",
+		},
+		schema: [],
+		type: "problem",
+	},
+	create: (context) => {
+		return {
+			"ExpressionStatement > CallExpression > MemberExpression[property.name='subscribe']":
```
Comment (ota-meshi) [inline src/rules/no-ignored-unsubscribe.ts:21]: Add `foo(a.subscribe)` to test case. I think there is probably a false positive.

You can fix it by adding a property like this:

```suggestion
			"ExpressionStatement > CallExpression > MemberExpression.callee[property.name='subscribe']":
```


https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IFwibm8tcmVzdHJpY3RlZC1zeW50YXhcIjogW1wiZXJyb3JcIiwgXCJFeHByZXNzaW9uU3RhdGVtZW50ID4gQ2FsbEV4cHJlc3Npb24gPiBNZW1iZXJFeHByZXNzaW9uLmNhbGxlZVtwcm9wZXJ0eS5uYW1lPSdzdWJzY3JpYmUnXVwiXSAqL1xuXG5mb28oYS5zdWJzY3JpYmUpXG5hLnN1YnNjcmliZSgpIiwib3B0aW9ucyI6eyJlbnYiOnsiZXM2Ijp0cnVlfSwicnVsZXMiOnt9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCJ9fX0=

**Review by ota-meshi — CHANGES_REQUESTED** [review]

**Comment by moufmouf** [comment]
Hey @ota-meshi!

Thanks a lot for this very quick and thorough review. I'm impressed!
I applied your suggestions and made the requested changes. Let me know if something is missing!

**Review by ota-meshi — APPROVED** [review]
LGTM! Thanks for suggesting and implementing the rule!

**Pull request merged**